### PR TITLE
[#IOPID-581] add subnet to downstream components

### PR DESCRIPTION
**Pull request by arcogabbo**
This PR depends on #678 for the subnet creation

### List of changes
* added the azure_subnet of `ioweb-profile-fn` in `core`, `ioweb-common` and `citizen-auth-app` domains (functions_app, ioweb hub-spid-login instance and functions_fast_login are all downstream micro-services called by the new azure function)
* added the azure_subnet of `functions_fast_login` in `core` domain(appbackendli will also be a downstream component)

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/core/function_app.tf [hunk 1]
```diff
@@ -160,6 +160,12 @@ module "app_snet" {
   }
 }
 
+data "azurerm_subnet" "ioweb_profile_snet" {
+  name                 = format("%s-ioweb-profile-snet", local.project)
```
Comment (gquadrati) [inline src/core/function_app.tf:164]: location `weu` is missing here

File: src/domains/citizen-auth-app/01_network.tf [hunk 2]
```diff
@@ -69,6 +69,12 @@ data "azurerm_subnet" "app_backend_l2_snet" {
   resource_group_name  = local.vnet_common_resource_group_name
 }
 
+data "azurerm_subnet" "ioweb_profile_snet" {
+  name                 = format("%s-ioweb-profile-snet", local.product)
```
Comment (gquadrati) [inline src/domains/citizen-auth-app/01_network.tf:73]: location `weu` is missing here

File: src/domains/ioweb-common/01_network.tf [hunk 3]
```diff
@@ -21,6 +21,12 @@ data "azurerm_subnet" "apim_v2_snet" {
   resource_group_name  = local.vnet_common_resource_group_name
 }
 
+data "azurerm_subnet" "ioweb_profile_snet" {
+  name                 = format("%s-ioweb-profile-snet", local.product)
```
Comment (gquadrati) [inline src/domains/ioweb-common/01_network.tf:25]: ditto

**Review by gquadrati — CHANGES_REQUESTED** [review]
We also need to include the `function-fast-login` subnet among the allowed subnets for `appbackendli`

File: src/domains/ioweb-common/01_network.tf (hunk 3 above)
Comment (arcogabbo) [reply src/domains/ioweb-common/01_network.tf]: fixed in https://github.com/pagopa/io-infra/pull/679/commits/68d692316214b8cd97fc28da3195bdf7f818de2c

File: src/domains/citizen-auth-app/01_network.tf (hunk 2 above)
Comment (arcogabbo) [reply src/domains/citizen-auth-app/01_network.tf]: fixed in https://github.com/pagopa/io-infra/pull/679/commits/68d692316214b8cd97fc28da3195bdf7f818de2c

File: src/core/function_app.tf (hunk 1 above)
Comment (arcogabbo) [reply src/core/function_app.tf]: fixed in https://github.com/pagopa/io-infra/pull/679/commits/68d692316214b8cd97fc28da3195bdf7f818de2c

**Comment by arcogabbo** [comment]
> We also need to include the `function-fast-login` subnet among the allowed subnets for `appbackendli`

fixed in https://github.com/pagopa/io-infra/pull/679/commits/f307bbb962c8d2f040996b1b5764d25fac0792cb

**Comment by arcogabbo** [comment]
> We also need to include the `function-fast-login` subnet among the allowed subnets for `appbackendli`

fixed in https://github.com/pagopa/io-infra/pull/679/commits/92aabb3b5e27640fa891b2a5125ca5bfd9dddb61

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Review by gquadrati — APPROVED** [review]

**Pull request merged**